src/server/frontend_wayland/window_wl_surface_role.cpp

```diff
@@ -383,12 +394,47 @@ mir::shell::SurfaceSpecification& mf::WindowWlSurfaceRole::spec()
     return *pending_changes;
 }
 
+void mf::WindowWlSurfaceRole::apply_pending_size_bounds()
+{
+    if (weak_scene_surface.lock())
+    {
+        if (pending_changes)
+        {
+
+#define SET_IF_CHANGED(pending, committed)  \
+    if (pending)                            \
+    {                                       \
+        if (pending.value() == committed)   \
+            pending.consume();              \
+        else                                \
+            committed = pending.value();    \
+    }
+
+            SET_IF_CHANGED(pending_changes->min_width,  committed_min_size.width);
+            SET_IF_CHANGED(pending_changes->min_height, committed_min_size.height);
+            SET_IF_CHANGED(pending_changes->max_width,  committed_max_size.width);
+            SET_IF_CHANGED(pending_changes->max_height, committed_max_size.height);
+
+#undef set_if_changed
```

Comment:
I'm pretty sure this won't do what you want, as `set_if_changed` ≠ `SET_IF_CHANGED` :grin: